packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1203,7 +1217,29 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
             hideToolbar();
         }
       },
-    );
+    )..addAll(_textProcessingActionButtonItems);
+  }
+
+  List<ContextMenuButtonItem> get _textProcessingActionButtonItems {
+    final List<ContextMenuButtonItem> buttonItems = <ContextMenuButtonItem>[];
+    final SelectedContent? data = _selectable?.getSelectedContent();
+    if (data == null) {
+      return buttonItems;
+    }
+
+    for (final ProcessTextAction action in _processTextActions) {
+      buttonItems.add(ContextMenuButtonItem(
+        label: action.label,
+        onPressed: () async {
+          final String selectedText = data.plainText;
+          if (selectedText.isNotEmpty) {
+            await _processTextService.processTextAction(action.id, selectedText, true);
+            hideToolbar();
+          }
+        },
+      ));
+    }
+    return buttonItems;
```

Comment:
I want to point out that a lot of the code changes in this file are duplicating code that's now in EditableText, too, though I think that's inevitable for now. Hopefully we'll be able to deduplicate this as part of the larger project to unify the text selection code.

Reply:
> Hopefully we'll be able to deduplicate this as part of the larger project to unify the text selection code.

Very interesting, is there an issue to track this project?

Reply:
See https://github.com/flutter/flutter/issues/104547 and https://github.com/flutter/flutter/issues/129583, CC @Renzo-Olivares.